operator/v1/types.go

```diff
@@ -45,6 +46,30 @@ type ClusterManagerSpec struct {
 	// NodePlacement enables explicit control over the scheduling of the deployed pods.
 	// +optional
 	NodePlacement NodePlacement `json:"nodePlacement,omitempty"`
+
+	// DeployOption contains the options of deploying a cluster-manager
+	// Default mode is used if DeployOption is not set.
+	// +optional
+	DeployOption DeployOption `json:"deployOption,omitempty"`
+}
+
+type InstallMode string
+
+const (
+	InstallModeDefault  InstallMode = "Default"
+	InstallModeDetached InstallMode = "Detached"
+)
+
+type DeployOption struct {
+	// Mode can be Default or Detached.
+	// In Detached mode, only crd and configurations are installed on the hub. Controllers are run in another cluster (defined as management cluster) and connect to the hub with the kubeconfig in secret of "external-hub-kubeconfig".
+	// The purpose of Detached mode is to give it more flexibility, for example we can install a hub on a cluster with no worker nodes, meanwhile running all deployments on another more powerful cluster.
+	// Also "external-hub-kubeconfig" should contain the kubeconfig with admin permission of hub-cluster.
```

Comment:
Does the `external-hub-kubeconfig` still need admin permission?

Reply:
I suppose yes. With admin permission do no harm since we will use kubeconfig generated by SA syncer.

Reply:
I think a detailed permission description is more friendly for users. We describe the minimum permission requirement, users can use admin as well.

Reply:
Yes, that makes sense. I add a description to emphasize the kubeconfig should have appropriate permission here. In the registration-operator, I will add a makefile command to generate an external-hub-config with a correct role and rolebinding as same as the default mode: https://github.com/open-cluster-management-io/registration-operator/blob/main/deploy/cluster-manager/config/rbac/cluster_role.yaml

Reply:
I think neither "appropriate authority" nor "management authority" is a proper description. but the detailed permission needs a lot of description, so I don't know how to describe it here either. @qiujian16 WDYT?